### ISSUE TITLE
Tests fail on OpenJDK 12 due to 'Failed to find the "modifiers" field in method setInternalState.'

### ISSUE THREAD

**Issue by krichter722** [checkstyle/checkstyle#7033]
Checkstyle build by `mvn install` fails on OpenJDK 12 due to

```
java.lang.RuntimeException: Internal error: Failed to find the "modifiers" field in method setInternalState.

	at org.powermock.reflect.internal.WhiteboxImpl.sedModifiersToField(WhiteboxImpl.java:2356)
	at org.powermock.reflect.internal.WhiteboxImpl.removeFinalModifierIfPresent(WhiteboxImpl.java:2320)
	at org.powermock.reflect.internal.WhiteboxImpl.setField(WhiteboxImpl.java:2306)
	at org.powermock.reflect.internal.WhiteboxImpl.setInternalState(WhiteboxImpl.java:304)
	at org.powermock.reflect.Whitebox.setInternalState(Whitebox.java:174)
	at com.puppycrawl.tools.checkstyle.api.FileContentsTest.testGetJavadocBefore(FileContentsTest.java:240)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
Caused by: java.lang.NoSuchFieldException: modifiers
	at java.base/java.lang.Class.getDeclaredField(Class.java:2416)
	at org.powermock.reflect.internal.WhiteboxImpl.sedModifiersToField(WhiteboxImpl.java:2350)
	... 32 more
```

I propose exposing `FileContents.javadocComments` with a package-protected getter which is used instead of the Powermock's `Whitebox.setInternalState`.

I think @rnveach already did some research on this in https://github.com/powermock/powermock/issues/939. I however am convinced that sacrificing the hidden field for easier tests is the way to go.

The tests are currently excluded which is not ideal because OpenJDK 12 is GA.

I can open a PR for that.

**Comment by romani** [comment]
@krichter722 , sorry ....., no "mvn", "gradle" .... only CLI is approved way to report issue.
please strictly follow issue template. If you can not follow it, most likely this is not a good project to report issue.

**Comment by krichter722** [comment]
@romani Build and development issues can't be reported using the template because nothing in it applies to this class of issues.

> If you can not follow it, most likely this is not a good project to report issue.

Do you mean that there's a separate project where build and development issues are reported or that you assume that the report doesn't meet the quality standards if it doesn't follow the template and therefore is not "good". The latter doesn't apply as explained.

I just opened this issue because having an "approved" is precondition for opening a PR. I can do that without the issue as well.

**Comment by krichter722** [comment]
@romani I think I understand: You think that I'm referring to a project of mine where I run `mvn install` without giving a reproducer. I'm referring to checkstyle as project which I'm building.

**Comment by romani** [comment]
yes, I misread your descrition.

Yes, build problems of chekcstyle is valid issue, thanks for reporting it.

**Comment by krichter722** [comment]
You excluded the failing tests in #6687.

**Comment by rnveach** [comment]
@krichter722 
> sacrificing the hidden field for easier tests is the way to go

Which field in our project are you talking about?
We generally do not like to loosen modifiers to make testing easier. Tests are in the same package as class, so package-private or protected may be more acceptable.

**Comment by romani** [comment]
I see nuance - https://github.com/checkstyle/checkstyle/blob/master/.ci/travis/travis.sh#L112
we skip some tests.. postponed resolving of them

PR is welcome.

**Comment by krichter722** [comment]
@rnveach

> I propose exposing FileContents.javadocComments with a package-protected getter which is used instead of the Powermock's Whitebox.setInternalState.

**Comment by rnveach** [comment]
Any tests that do not require powermock anymore should be moved to non-powermock test class.
Powermock was separated out after seeing it caused issues with pitest.

**Comment by krichter722** [comment]
I don't seem to be able to fix

```
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.204 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.internal.powermock.ImportControlLoaderPowerTest
[ERROR] testInputStreamThatFailsOnClose(com.puppycrawl.tools.checkstyle.internal.powermock.ImportControlLoaderPowerTest)  Time elapsed: 1.006 s  <<< FAILURE!
org.mockito.exceptions.verification.TooManyActualInvocations: 

inputStream.close();
Wanted 1 time but was 2 times.
	at com.puppycrawl.tools.checkstyle.internal.powermock.ImportControlLoaderPowerTest.testInputStreamThatFailsOnClose(ImportControlLoaderPowerTest.java:69)
```

Do you have an idea why the number of invocations changed?

**Comment by romani** [comment]
> I propose exposing FileContents.javadocComments with a package-protected getter

it is API class
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java
it is absolute last resort, to use it, we need to be sure that there is no other way out.

**Comment by romani** [comment]
failures of FileTextTest on jdk12: 
```
-------------------------------------------------------------------------------
Test set: com.puppycrawl.tools.checkstyle.api.FileTextTest
-------------------------------------------------------------------------------
Tests run: 9, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.003 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.api.FileTextTest
testFindLineBreaksCache(com.puppycrawl.tools.checkstyle.api.FileTextTest)  Time elapsed: 0.001 s  <<< ERROR!
java.lang.RuntimeException: Internal error: Failed to find the "modifiers" field in method setInternalState.
	at com.puppycrawl.tools.checkstyle.api.FileTextTest.testFindLineBreaksCache(FileTextTest.java:158)
Caused by: java.lang.NoSuchFieldException: modifiers
	at com.puppycrawl.tools.checkstyle.api.FileTextTest.testFindLineBreaksCache(FileTextTest.java:158)

testFindLineBreaks(com.puppycrawl.tools.checkstyle.api.FileTextTest)  Time elapsed: 0.001 s  <<< ERROR!
java.lang.RuntimeException: Internal error: Failed to find the "modifiers" field in method setInternalState.
	at com.puppycrawl.tools.checkstyle.api.FileTextTest.testFindLineBreaks(FileTextTest.java:140)
Caused by: java.lang.NoSuchFieldException: modifiers
	at com.puppycrawl.tools.checkstyle.api.FileTextTest.testFindLineBreaks(FileTextTest.java:140)
```

**Comment by krichter722** [comment]
@romani The changes are included in #7035 where I incorporated you requested changes. Github doesn't recognize them afaik, but you can see the changes in the diff. #7035 should be merged now because it becomes hard to keep track of all these issues and pull requests.

**Comment by romani** [comment]
@krichter722 ,

Thanks a lot to keep interest to issue.
As I already mentioned in PR several times, I completely disagree with proposed fix. You insist on your fix, we are in conflict. I do not close your PR, only because I am not 100% in success of my proposal. I can not dedicate enough time to tackle all items of issue. So I try to tackle it test by test.
You can see my second step at #7213.

If in some cases you are right, I will reuse your code. 
It would make it quicker, if you cooperate on approach  I suggest, or give more reasons why my approach does not work.

I need urgently switch to eclipse-cs project to unblock it's development, so I will be slow in library project. Your help is in high demand.

**Comment by krichter722** [comment]
@romani The conflict is resolved. I'll follow your approach and made the changes in the PR. Github doesn't display it correctly in the conversation, see the diff for reference. In doubt, merge #7213 and I'll rebase on it.

**Comment by romani** [comment]
Fix is merged
